ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/S3CompatibleIntegrationForm.tsx

```diff
@@ -0,0 +1,486 @@
+/* eslint-disable no-void */
+import React, { ReactElement } from 'react';
+import {
+    Checkbox,
+    Form,
+    FormSelect,
+    PageSection,
+    Popover,
+    TextInput,
+} from '@patternfly/react-core';
+import { SelectOption } from '@patternfly/react-core/deprecated';
+import { HelpIcon } from '@patternfly/react-icons';
+import * as yup from 'yup';
+
+import { BackupIntegrationBase } from 'services/BackupIntegrationsService';
+
+import usePageState from 'Containers/Integrations/hooks/usePageState';
+import FormMessage from 'Components/PatternFly/FormMessage';
+import FormCancelButton from 'Components/PatternFly/FormCancelButton';
+import FormTestButton from 'Components/PatternFly/FormTestButton';
+import FormSaveButton from 'Components/PatternFly/FormSaveButton';
+import SelectSingle from 'Components/SelectSingle';
+import useIntegrationForm from '../useIntegrationForm';
+import { IntegrationFormProps } from '../integrationFormTypes';
+
+import IntegrationFormActions from '../IntegrationFormActions';
+import FormLabelGroup from '../FormLabelGroup';
+import ScheduleIntervalOptions from '../FormSchedule/ScheduleIntervalOptions';
+import ScheduleWeeklyOptions from '../FormSchedule/ScheduleWeeklyOptions';
+import ScheduleDailyOptions from '../FormSchedule/ScheduleDailyOptions';
+
+const urlStyles = [
+    {
+        label: 'Path',
+        value: 'S3_URL_STYLE_PATH',
+    },
+    {
+        label: 'Virtual hosted',
+        value: 'S3_URL_STYLE_VIRTUAL_HOSTED',
+    },
+];
+
+export type S3CompatibleIntegration = {
+    s3compatible: {
+        bucket: string;
+        objectPrefix: string;
+        endpoint: string;
+        region: string;
+        accessKeyId: string;
+        secretAccessKey: string;
+        urlStyle: 'S3_URL_STYLE_PATH' | 'S3_URL_STYLE_VIRTUAL_HOSTED';
+    };
```

Comment:
Although best to follow precedent of redundant declaration in most integrationforms, it is future tech debt task for UI team to import type from frontend proto (or even better, service) file as source of truth.